scripts/load_meeting_prot.py

```diff
@@ -14,125 +17,119 @@
-        elif not row[col+2].strip()=='' and FC:
-            data[proj]['flowcells'][FC].append(row[col+2])
+    for row in content:
+        col_A = row[0].strip()
+        col_B = row[1].strip()
+        col_C = row[2].strip()
+        if col_A and col_A != 'Ongoing deliveries':
+            proj_name = col_A
+            data[proj_name] = {'info': col_B, 'flowcells':{}}
+        elif col_B:
+            flow_cell = col_B
+            data[proj_name]['flowcells'][flow_cell] = [col_C]
+        elif col_C and flow_cell:
+            data[proj_name]['flowcells'][flow_cell].append(col_C)
     return data
 
 def sort_by_name(namelist):
     "Sorts dict alphabeticly by project sure name"
-    name_dict={}
-    for proj in namelist:
-        name_dict[proj]=proj.split('.')[1].strip()
+    name_dict = {}
+    for proj_name in namelist:
+        name_dict[proj_name] = proj_name.split('.')[1].strip()
     sorted_name_dict = sorted(name_dict.iteritems(), key=operator.itemgetter(1))
     return sorted_name_dict
 
-def pars_file(file, ongoing_deliveries, content):
+def merge_info_from_file_and_wsheet(trello_dump, old_wsheet_content):
     """parses output from script "update_checklist.py", which is loading runinfo
     from the trello board. ongoing_deliverues is a dict of info from old meeting
-    that will be merged with the new info feched from the trello."""
+    that will be merged with the new info feched from trello."""
+    ongoing_deliveries = get_meeting_info_from_wsheet(old_wsheet_content)
+    f = open(trello_dump,'r')
+    content = f.readlines()
     coming_deliveries = {}
-    f=open(file,'r')
-    content=f.readlines()
-    nlines=len(content)
-    data = coming_deliveries
-    i=0
-    proj=None
-    while i < nlines:
-        row = content[i]
-        if row.strip() == "Ongoing":
-            data = ongoing_deliveries
-        if not row.strip()=='':
-            isproj=row.split()[0]+'XXX'
-            if isproj[1]=='.' or isproj[2]=='.':
-                proj = row.split()[0].strip()
-                if not proj in data.keys():
-                    data[proj]={'info':' '.join(row.split()[1:]), 'flowcells':{}}
-            elif proj and row.split()[0][0]!='1':
-                if data[proj].has_key('info'):
-                    if row.strip() not in data[proj]['info']:
-                        data[proj]['info'] = data[proj]['info']+' - '+row
-            elif proj and row.split()[0][0]=='1':
-                if not data[proj]['flowcells'].has_key(row.strip()):
-                    data[proj]['flowcells'][row.strip()]=[]
-                proj=None
-            i=i+1
-        else:
-            i=i+1
-    return coming_deliveries, ongoing_deliveries
+    dict_holder = coming_deliveries
+    proj_name = None
+    for row in content:
+        row = row.strip()
+        if row == "Ongoing":
+            dict_holder = ongoing_deliveries
+        if row:
+            if len(row) > 2 and row[1] == '.' or row[2] == '.':
```

Comment:
I think you're right. I you want to use complex conditions, parenthesis are
always nice to help the reader understand what is happening

On Wed, Jan 28, 2015 at 10:17 AM, Mario Giovacchini <
notifications@github.com> wrote:

> In scripts/load_meeting_prot.py
> <https://github.com/SciLifeLab/scilifelab/pull/461#discussion_r23673921>:
>
> > -            elif proj and row.split()[0][0]=='1':
> > -                if not data[proj]['flowcells'].has_key(row.strip()):
> > -                    data[proj]['flowcells'][row.strip()]=[]
> > -                proj=None
> > -            i=i+1
> > -        else:
> > -            i=i+1
> > -    return coming_deliveries, ongoing_deliveries
> > +    dict_holder = coming_deliveries
> > +    proj_name = None
> > +    for row in content:
> > +        row = row.strip()
> > +        if row == "Ongoing":
> > +            dict_holder = ongoing_deliveries
> > +        if row:
> > +            if len(row) > 2 and row[1] == '.' or row[2] == '.':
>
> I believe Python just evaluates left to right until it logically doesn't
> need to anymore. So True and False or True would evaluate all the way to
> the last True, whereas True or False and False would return True and
> ignore anything after the or. Someone can correct me if I'm wrong.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciLifeLab/scilifelab/pull/461/files#r23673921>.
>